src/engine/drpc_client.c

```diff
@@ -385,12 +385,18 @@ ds_pool_find_bylabel(d_const_string_t label, uuid_t pool_uuid,
 		D_GOTO(out_resp, rc = frsp->status);
 	}
 
+	rc = uuid_parse(frsp->uuid, pool_uuid);
+	if (rc != 0) {
+		D_ERROR("Unable to parse pool UUID %s: "DF_RC"\n", frsp->uuid,
+			DP_RC(rc));
+		D_GOTO(out_resp, rc = -DER_IO);
```

Comment:
minor nit: Is -DER_IO the right error here? Seems like maybe -DER_INVAL might be better?

Reply:
Oh, hah. I see the @liw requested a change from -DER_INVAL.